specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/preview/2024-09-01-preview/appconfiguration.json

```diff
@@ -0,0 +1,3168 @@
+          "type": "string",
+          "enum": [
+            "Primary",
+            "Secondary"
+          ],
+          "x-ms-enum": {
+            "name": "sasKind",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "SignedParameters": {
+      "description": "The parameters to define the data plane resource scope that the SAS token is authorized to access.",
+      "type": "object",
+      "properties": {
+        "key": {
+          "description": "A filter used to match keys. Syntax reference:\nhttps://aka.ms/azconfig/docs/keyvaluefiltering",
+          "type": "string"
+        },
+        "label": {
+          "description": "A filter used to match labels. Syntax reference:\nhttps://aka.ms/azconfig/docs/keyvaluefiltering",
+          "type": "string"
+        },
+        "tags":{
+          "description": "Am array of tag filters used to match tags.",
+          "type": "array",
+          "items": {
+            "description": "A filter used to match tag.",
+            "type": "string"
+          }
+        },
+        "name":{
+          "description": "The name of the snapshot. This is only applicable when the resourceType is snapshot.",
+          "type": "string"
+        }
+      }
+    },
+    "SasTokenGenerateResult": {
+      "description": "The result of a request to generate a SAS token.",
+      "type": "object",
+      "properties": {
+        "tokenValue":{
+          "description": "The value of the SAS token.",
+          "type": "string",
+          "format": "password",
+          "x-ms-secret": true,
+          "readOnly": true
+        },
+        "resourceType": {
+          "description": "The data plane resource type that the SAS token is authorized to access.",
+          "type": "string",
+          "readOnly": true
+        },
+        "signedParameters": {
+          "description": "The parameters to define the data plane resource scope.",
+          "$ref": "#/definitions/SignedParameters",
+          "readOnly": true
+        },
+        "exp":{
+          "description": "The time that the SAS token expires in the Universal ISO 8601 DateTime format. Max allowed expiration is 1 year from the time of token creation.",
+          "type": "string",
+          "format": "date-time",
+          "readOnly": true
+        },
+        "cch":{
+          "description": "Time (in seconds) for which the response may be cached by clients.",
+          "type": "number"
+        },
+        "sasKind":{
+          "description": "The kind of the SAS token.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "ResetSasParameters": {
+      "description": "Parameters used for resetting SAS token.",
+      "required": [
+        "sasKind"
+      ],
+      "type": "object",
+      "properties": {
+        "sasKind":{
+          "description": "The kind of the SAS token.",
+          "type": "string",
+          "enum": [
+            "Primary",
```

Comment:
Are these case sensitive? They shouldnt need to be.

Reply:
I think generally we make the value of enum that modeled as string in this case sensitive format. For example, "Enabled", "Disabled" for "publicNetworkAccess"

Reply:
For resourceType "kv" and "snapshot", if following the same convention, maybe I should also update them to "Kv" and "Snapshot"

Reply:
So what happens if someone provides 'enabled' for "publicNetworkAccess"? Do we fail the request?

Reply:
No, we didn't fail the request, the allowed values check of "publicNetworkAccess" ignores cases. But we did it differently for different properties. For "dataplaneproxy.authenticationMode", the check doesn't ignore case and we will fail the request.

Reply:
Ok, so the case sensitivity is decided in the implementation. We can ignore the case for SAS parameters in the RP implementation.